src/Hpack.hs

```diff
@@ -192,7 +194,7 @@ hpackResult = hpackResultWithVersion version
 
 hpackResultWithVersion :: Version -> Options -> IO Result
 hpackResultWithVersion v (Options options force generateHashStrategy toStdout) = do
-  DecodeResult pkg (lines -> cabalVersion) cabalFileName warnings <- readPackageConfig options >>= either die return
+  DecodeResult pkg (lines -> cabalVersion) cabalFileName warnings <- readPackageConfig options >>= either throwIO return
```

Comment:
Instead of using `throwIO`, my suggestion is:

1. change the return type of `hpackResultWithVersion` to `Either`
1. export a new primitive `hpackFoo = hpackResultWithVersion version` (that's what stack is going to use; try to find a better name)
1. change `hpack` and `hpackResult` so that they retain their current behavior.

Reply:
First of all an apology - I did not read your initial comment on the related issue closely enough - sorry. I now understand better. Second, I think this is now done. I called the new primative `hpackResultWithException`.

Reply:
That's OK for now, even though I have the feeling that there will be no exception once we are done with this. So we may end up renaming this to `hpackResultWithError` or something. But let's defer that for now.